### PR TITLE
Check formatting as part of the CI

### PR DESCRIPTION
**Pull request by flomnes**
Succeed if all source files are well formated. Fail otherwise, and display ill-formatted files.

### CODE REVIEW

File: .github/workflows/clang-format.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+on:
+  push:
+    branches:
+      - develop
+      - feature/*
+      - features/*
+      - fix/*
+      - release/*
+      - doc/*
+      - dependabot/*
```
Comment (a-zakir) [inline .github/workflows/clang-format.yml]:
```suggestion
```

File: .github/workflows/clang-format.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: clang-format
+        run: cd src && ./format-code.sh
+
+      - name: git diff
+        run: |
+          DIFF=`git status --porcelain`
+          if [[ $DIFF ]]; then
+          echo "The following files are not well formatted"
+          echo "$DIFF"
```
Comment (a-zakir) [inline .github/workflows/clang-format.yml:32]: upload diff file?

Comment (flomnes) [reply .github/workflows/clang-format.yml:32]: why ?

File: .github/workflows/clang-format.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Check formatting
```
Comment (a-zakir) [inline .github/workflows/clang-format.yml]:
```suggestion
name: Check cpp formatting
```

Comment (a-zakir) [reply .github/workflows/clang-format.yml]: or clang-format

File: .github/workflows/clang-format.yml (hunk 2 above)
Comment (a-zakir) [reply .github/workflows/clang-format.yml:32]: to dl & apply the patch ;)

**Comment by a-zakir** [comment]
i'll consider formatting changed files only

**Comment by a-zakir** [comment]
> ~i'll consider formatting changed files only~
this Pr is definitely an enhancement

**Review by a-zakir — APPROVED** [review]

**Pull request merged**